modal-js/src/sandbox.ts

```diff
@@ -164,6 +164,37 @@ export class Sandbox {
     return new Sandbox(sandboxId);
   }
 
+  /** Get a running Sandbox by name from a deployed App.
+   *
+   * Raises a NotFoundError if no running Sandbox is found with the given name.
+   * A Sandbox's name is the `name` argument passed to `App.createSandbox`.
+   *
+   * @param appName - Name of the deployed app
+   * @param name - Name of the sandbox
+   * @param environment - Optional override for the environment
+   * @returns Promise that resolves to a Sandbox
+   */
+  static async fromName(
+    appName: string,
+    name: string,
+    environment?: string,
+  ): Promise<Sandbox> {
+    try {
+      const resp = await client.sandboxGetFromName({
+        sandboxName: name,
+        appName,
+        environmentName: environmentName(environment),
+      });
+      return new Sandbox(resp.sandboxId);
+    } catch (err) {
+      if (err instanceof ClientError && err.code === Status.NOT_FOUND)
+        throw new NotFoundError(
+          `Sandbox with name '${name}' not found in app '${appName}'`,
```

Comment:
Just noting that [we return an error from the server](https://github.com/modal-labs/modal/blob/main/server/modal_server/objects/sandbox.py#L807) that mentions the caveat that Sandboxes can only be looked up by name from _deployed_ apps. This might not be an issue from non-Python SDKs though? Can you even run an ephemeral app with the non-Python SDKs?